ui/panels-plugin/src/plugins/scatterplot/ScatterChartPanel.tsx

```diff
@@ -0,0 +1,165 @@
+ * data formatting is also dictated in this component. Formatting includes
+ * datapoint size and color.
+ *
+ * Documentation for data structures accepted by Apache ECharts:
+ *  https://echarts.apache.org/handbook/en/concepts/dataset
+ *
+ * Examples for scatter chart formatting in Apache ECharts:
+ *  https://echarts.apache.org/examples/en/index.html#chart-type-scatter
+ *
+ * @returns a `ScatterPlot` component that contains an EChart which will handle
+ * visuzliation of the data.
+ */
+export function ScatterChartPanel(props: ScatterChartPanelProps) {
+  const { contentDimensions } = props;
+  const { queryResults: traceResults, isLoading: traceIsLoading } = useDataQueries('TraceQuery');
+
+  // Generate dataset
+  // Transform Tempo API response to fit 'dataset' structure from Apache ECharts
+  // https://echarts.apache.org/handbook/en/concepts/dataset
+  const dataset = useMemo(() => {
+    const traceData = traceResults[0]?.data;
+    if (traceIsLoading || traceData === undefined) {
+      return [];
+    }
+    const dataset = [];
+    for (const result of traceResults) {
+      if (traceIsLoading || traceData === undefined) {
+        return [];
+      }
+      if (result.isLoading || result.data === undefined) continue;
+      const dataSeries = result.data.traces.map((trace) => {
+        const newTraceValue: EChartTraceValue = {
+          ...trace,
+          startTime: new Date(trace.startTimeUnixMs), // convert unix epoch time to Date
+        };
+        return newTraceValue;
+      });
+      dataset.push({
+        source: dataSeries,
+      });
+    }
+    return dataset;
+  }, [traceIsLoading, traceResults]);
+
+  // Error check: specify an alert if no traces are returned from the query
+  const traceData = traceResults[0]?.data;
+  if (!traceIsLoading && traceData?.traces.length === 0) {
+    const query = traceData?.metadata?.executedQueryString;
+    return generateErrorAlert(`No traces found for the query : " ${query} " .`);
+  }
+
+  // Formatting for the dataset
+  // 1. Map x,y coordinates
+  // 2. Datapoint size corresponds to the number of spans in a trace
+  // 3. Color datapoint red if the trace contains an error
+  const seriesTemplate: SeriesOption = {
+    type: 'scatter',
+    encode: {
+      // Map to x-axis.
+      x: 'startTime',
+      // Map to y-axis.
+      y: 'durationMs',
+    },
+    symbolSize: function (data) {
+      // Changes datapoint to correspond to number of spans in a trace
+      const scaleSymbolSize = 10;
+      return data.spanCount * scaleSymbolSize;
+    },
+    itemStyle: {
+      color: function (params) {
+        const traceData: EChartTraceValue = params.data as EChartTraceValue;
+        // If the trace contains an error, color the datapoint in red
+        if (traceData.errorCount !== undefined && traceData.errorCount > 0) {
+          return 'red';
+        }
+        // Else return default color
+        const defaultColor = '#56B4E9';
```

Comment:
chartsTheme is nice to leverage for these types of colors since it allows embed users to customize their palette, fonts, etc. ([ex of defaultColor ](https://github.com/perses/perses/blob/072cb4f8aa8ad6a36a014c490cd357c31b87fdba/ui/panels-plugin/src/plugins/stat-chart/utils/data-transform.ts#L25)for thresholds)

Reply:
Absolutely, agree. Great catch; this is updated in the latest commit.

Reply:
Awesome and you can always add new colors too if what’s there isn’t enough